recipe/entry_point.py

```diff
@@ -12,6 +13,40 @@ def map(self, func, *iterables):
             for thing in iterable:
                 yield func(thing)
 
+# This might be None!
+CPU_COUNT = os.cpu_count()
+
+# See validation results for magic number of 3
+# https://dholth.github.io/conda-benchmarks/#extract.TimeExtract.time_extract?conda-package-handling=2.0.0a2&p-format='.conda'&p-format='.tar.bz2'&p-lang='py'&x-axis=format
+DEFAULT_NUM_WORKERS = 1 if not CPU_COUNT else min(3, CPU_COUNT)
+
+def validate_max_workers(num_str: str) -> int:
+    """Converts a string representing the max number of workers to an integer
+    while performing validation checks; raises ValueError if anything fails."""
+
+    ERROR_MSG = "Max workers must be integer between 1 and the CPU count of the machine."
+
+    try:
+        num = int(num_str)
+    except ValueError as exc:
+        raise ValueError(ERROR_MSG) from exc
+
+    if (num < 1):
+        raise ValueError(ERROR_MSG)
+
+    # cpu_count can return None, so skip this check if that happens
+    if CPU_COUNT:
+
+        # See Windows notes for magic number of 61
+        # https://docs.python.org/3/library/concurrent.futures.html#processpoolexecutor
+        max_cpu_num = min(CPU_COUNT, 61) if (os.name == "nt") else CPU_COUNT
+
+        if (num > max_cpu_num):
+            raise ValueError(ERROR_MSG)
+
+    return num
+
+
```

Comment:
I might be misunderstanding but in practice, the allowed CPU values are 1, 2 o 3, with three being the most performant according to the benchmarks.

Does it make sense to allow values greater than 3?

If not, we can just indicate that the validation logic is `int(x) in range(1,4)`. And the default is `min(3, os.cpu_count() or 1)`. That'd simplify this block quite a bit.

Reply:
I'm OK with limiting it to `min(CPU_COUNT, 3)` as long as I can also force it to 1 if needed. The big thing I need is just to make sure that it's not greedily consuming all available CPU threads, because that does not play nicely with corporate anti-virus/anti-malware software and then the constructor-built installer crashes. However, I can imagine that on a beefy Windows server with a couple of Xeons, the benchmarking results might indeed favor more than three threads, but fewer than all of them. I'm happy to play around with this after the holidays, and this switch would make it pretty easy for me to run some tests. Can always limit it to 3 in a future release if there's no good reason to have more.

Reply:
Btw, the Python docs say:

> os.cpu_count()
> Return the number of CPUs in the system. Returns None if undetermined.
> 
> This number is not equivalent to the number of CPUs the current process can use. The number of usable CPUs can be obtained with len(os.sched_getaffinity(0))

Should we use the 'usable' ones?

Reply:
No, too complicated

Reply:
No, because `os.sched_getaffinity` is not supported on Windows.

Reply:
(we never get to have the good toys)